apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Extensions/MarketParticipantClientExtensions.cs

```diff
@@ -62,4 +64,36 @@ internal static async Task<IEnumerable<GridAreaDto>> GetGridAreasAsync(
                     };
             });
     }
+
+    internal static async Task<IEnumerable<GridAreaDto>> GetRelevantGridAreasAsync(
+        this IMarketParticipantClient_V1 client,
+        Guid actorId,
+        Interval period,
+        CancellationToken cancellationToken = default)
+    {
+        var actor = await client.ActorGetAsync(actorId).ConfigureAwait(false);
+
+        var gridAreas = await client.GetGridAreasAsync();
+        var filteredByDateGridAreas = gridAreas.Where(ga => DoDatesOverlap(ga, period.Start.ToDateTimeOffset(), period.End.ToDateTimeOffset()));
+
+        var actorGridAreaIds = actor.MarketRole.GridAreas.Select(ga => ga.Id);
+        var relevantGridAreas = filteredByDateGridAreas.Where(ga => actorGridAreaIds.Contains(ga.Id));
+
+        return relevantGridAreas;
+    }
+
+    private static bool DoDatesOverlap(GridAreaDto gridArea, DateTimeOffset startDate, DateTimeOffset endDate)
+    {
+        var convertedStartDate = TimeZoneInfo.ConvertTimeBySystemTimeZoneId(startDate, "Romance Standard Time");
```

Comment:
Suggestion: Can this not be used https://nodatime.org/3.2.x/api/NodaTime.DateInterval.html#NodaTime_DateInterval_Contains_NodaTime_DateInterval_ ?